jerry-core/jerry.c

```diff
@@ -1841,81 +1824,117 @@ snapshot_add_compiled_code (ecma_compiled_code_t *compiled_code_p, /**< compiled
   if ((snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG) > 0xffffu)
   {
     snapshot_error_occured = true;
-    return;
+    return 0;
   }
 
-  snapshot_last_compiled_code_offset = snapshot_buffer_write_offset;
+  uint16_t start_offset = (uint16_t) (snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG);
+  ecma_compiled_code_t *copied_compiled_code_p;
 
-  compiled_code_map_entry_t *new_entry;
-  new_entry = (compiled_code_map_entry_t *) mem_heap_alloc_block (sizeof (compiled_code_map_entry_t));
+  copied_compiled_code_p = (ecma_compiled_code_t *) (snapshot_buffer_p + snapshot_buffer_write_offset);
 
-  if (new_entry == NULL)
+  if (!(compiled_code_p->status_flags & CBC_CODE_FLAGS_FUNCTION))
```

Comment:
At line 1847 we have `((re_compiled_code_t *) compiled_code_p)->pattern_cp`, which hints that `compiled_code_p` is a regex compiled code. However, it's true that `copied_compiled_code_p` is an `ecma_compiled_code_t`. Since it seems that `compiled_code_p` is only used once, leave it as is.